src/Data/Cursor/CursorBuilder.php

```diff
@@ -21,28 +21,34 @@ public function __construct( $compare = '>' ) {
 		$this->compare = $compare;
 		$this->fields  = [];
 	}
-
+	
 	/**
 	 * Add ordering field. The order you call this method matters. First field
 	 * will be the primary field and latters ones will be used if the primary
 	 * field has duplicate values
 	 *
-	 * @param string $key   database colum
-	 * @param string $value value from the current cursor
-	 * @param string $type  type cast
-	 * @param string $order custom order
+	 * @param string                  $key           database column
+	 * @param string                  $value         value from the current cursor
+	 * @param null                    $type          type cast
+	 * @param null                    $order         custom order
+	 * @param null | PostObjectCursor $object_cursor The PostObjectCursor class
 	 */
-	public function add_field( $key, $value, $type = null, $order = null ) {
+	public function add_field( $key, $value, $type = null, $order = null, $object_cursor = null ) {
 		/**
 		 * This only input for variables which are used in the SQL generation. So
 		 * escape them here.
 		 */
-		$this->fields[] = [
-			'key'   => esc_sql( $key ),
-			'value' => esc_sql( $value ),
-			'type'  => esc_sql( $type ),
-			'order' => esc_sql( $order ),
-		];
+		$this->fields[] = apply_filters(
+			'wpgraphql_cursor_ordering_field',
```

Comment:
@jacobarriola can we get a docblock over the cursor explaining the params?

something like: 

```
/**
 * Filters the field used for ordering when cursors are used for pagination
 * 
 * @param array $field The field key, value, type and order 
 * @param CursorBuilder $this The CursorBuilder class
 * @param mixed null | PostObjectCursor $cursor The PostObjectCursor class
 */

Reply:
@jacobarriola upon looking at this a bit more, we also need to make sure we're escaping _after_ the filter. 

So, I think we need to change this a bit more too. 

Something like: 

```php
$field = [
  'key'   => esc_sql( $key ),
  'value' => esc_sql( $value ),
  'type'  => esc_sql( $type ),
  'order' => esc_sql( $order ),
];

$filtered_field = apply_filters( 'wpgraphql_cursor_ordering_field', $field, $this, $object_cursor );
$clean_fields = [];
if ( ! empty( $field ) && is_array( $field ) ) {
  foreach( $field as $key => $value ) {
     $clean_fields[ $key ] = esc_sql( $value );
  }
}

$this->fields = $clean_fields;

```

Reply:
^ untested, just wrote here in Github as pseudo code, but might be legit 🤷‍♂️